velocity_modelling/write/hdf5.py

```diff
@@ -0,0 +1,283 @@
+            if lat_ind == ny - 1:
+                # Now that we have all slices, write the complete mesh data
+                mesh_group = f["mesh"]
+                mesh_group.create_dataset(
+                    "x", data=np.arange(partial_global_mesh.nx, dtype=np.float64)
+                )
+                mesh_group.create_dataset("y", data=np.arange(ny, dtype=np.float64))
+                # Depth is negative: -nz to 0
+                mesh_group.create_dataset(
+                    "z",
+                    data=np.arange(-1 * partial_global_mesh.nz, 0, dtype=np.float64),
+                )
+                mesh_group.create_dataset("lon", data=partial_global_mesh.lon)
+                mesh_group.create_dataset("lat", data=partial_global_mesh.lat)
+
+                # Mark file as complete
+                f.attrs["complete"] = True
+
+            # Write property data for this slice to make this Paraview compatible
+            # partial_global_qualities.vp is (nx, nz), need to transpose to (nz, nx) for (z, x)
+            # Dataset is (nz, ny, nx), so [:, lat_ind, :] expects (nz, nx)
+            vp_slice = np.transpose(partial_global_qualities.vp)  # (nx, nz) -> (nz, nx)
+            vs_slice = np.transpose(vs_data)  # (nz, nx)
+            rho_slice = np.transpose(partial_global_qualities.rho)  # (nz, nx)
+            inbasin_slice = np.transpose(partial_global_qualities.inbasin)  # (nz, nx)
+
+            # Verify shapes
+            expected_shape = (partial_global_mesh.nz, partial_global_mesh.nx)
+            if vp_slice.shape != expected_shape:
+                logger.log(
+                    logging.ERROR,
+                    f"Shape mismatch: vp_slice has shape {vp_slice.shape}, expected {expected_shape}",
+                )
+                raise ValueError(
+                    f"Shape mismatch: vp_slice has shape {vp_slice.shape}, expected {expected_shape}"
+                )
+
+            f["properties/vp"][:, lat_ind, :] = vp_slice
+            f["properties/vs"][:, lat_ind, :] = vs_slice
+            f["properties/rho"][:, lat_ind, :] = rho_slice
+            f["properties/inbasin"][:, lat_ind, :] = inbasin_slice
+
+            # Update progress attribute
+            f.attrs["last_slice_written"] = lat_ind
+
+        logger.log(logging.DEBUG, f"Written slice {lat_ind} to {hdf5_file}")
+    except OSError as e:
+        logger.log(logging.ERROR, f"Failed to update HDF5 file {hdf5_file}: {e}")
+        raise OSError(f"Failed to update HDF5 file {hdf5_file}: {str(e)}")
+    except Exception as e:
+        if isinstance(e, (SystemExit, KeyboardInterrupt)):
+            raise  # Re-raise critical exceptions
+        logger.log(logging.ERROR, f"Error writing to HDF5 file {hdf5_file}: {e}")
+        raise RuntimeError(f"Error writing to HDF5 file {hdf5_file}: {str(e)}")
+
+    # After writing all slices, create the XDMF file
+    if lat_ind == vm_params.get("ny", 0) - 1:
```

Comment:
Any reason not to use the `ny` variable here?
```suggestion
    if lat_ind == ny - 1:
```